examples/cpp/content_filter/CLIParser.hpp

```diff
@@ -0,0 +1,512 @@
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "invalid sample argument for " + arg + ": " + e.what());
+                        print_help(EXIT_FAILURE);
+                    }
+                    catch (const std::out_of_range& e)
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "sample argument out of range for " + arg + ": " + e.what());
+                        print_help(EXIT_FAILURE);
+                    }
+                }
+                else
+                {
+                    EPROSIMA_LOG_ERROR(CLI_PARSER, "missing argument for " + arg);
+                    print_help(EXIT_FAILURE);
+                }
+            }
+            else if (arg == "-i" || arg == "--interval")
+            {
+
+                if (i + 1 < argc)
+                {
+                    int input = std::stoi(argv[++i]);
+                    if (config.entity == CLIParser::EntityKind::PUBLISHER)
+                    {
+                        config.pub_config.interval = input;
+                    }
+                    else if (config.entity == CLIParser::EntityKind::SUBSCRIBER)
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "interval option can only be used with the Publisher");
+                        print_help(EXIT_FAILURE);
+                    }
+                    else
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "entity not specified for --sample argument");
+                        print_help(EXIT_FAILURE);
+                    }
+                }
+                else
+                {
+                    EPROSIMA_LOG_ERROR(CLI_PARSER, "missing argument for " + arg);
+                    print_help(EXIT_FAILURE);
+                }
+            }
+            else if (arg == "--reader-filters")
+            {
+                if (i + 1 < argc)
+                {
+                    if (config.entity == CLIParser::EntityKind::PUBLISHER)
+                    {
+                        config.pub_config.max_reader_filters = static_cast<uint16_t>(std::stoi(argv[++i]));
+                    }
+                    else if (config.entity == CLIParser::EntityKind::SUBSCRIBER)
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "--reader-filters is only valid for publisher entity");
+                        print_help(EXIT_FAILURE);
+                    }
+                    else
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "entity not specified for --reader-filters argument");
+                        print_help(EXIT_FAILURE);
+                    }
+                }
+                else
+                {
+                    EPROSIMA_LOG_ERROR(CLI_PARSER, "missing argument for " + arg);
+                    print_help(EXIT_FAILURE);
+                }
+            }
+            else if (arg == "--reliable")
```

Comment:
The `help message` is correct in the sense that it does not require an additional argument after `--reliable` which I think it is what makes sense, but here we are requiring it.
Same applies to `--transient-local`.
In this sense, the `help` message should state that the default values are `volatile` and `best_effort`, since the `--reliable` and `--transient-local` flags are meant to be put in bot publisher and subscriber to be coherent with both and with the other examples